test-frame-common/src/main/java/io/skodjob/testframe/resources/KubeResourceManager.java

```diff
@@ -50,148 +53,266 @@
 
+    private final Map<String, ClusterContext> clientCache = new ConcurrentHashMap<>();
+    private static final ThreadLocal<String> CURRENT_CLUSTER_CONTEXT = ThreadLocal.withInitial(() ->
+        TestFrameConstants.DEFAULT_CONTEXT_NAME);
     private static final ThreadLocal<ExtensionContext> TEST_CONTEXT = new ThreadLocal<>();
-    private static final Map<String, Stack<ResourceItem<?>>> STORED_RESOURCES = new LinkedHashMap<>();
 
-    private String storeYamlPath = null;
+    private static final KubeResourceManager INSTANCE = new KubeResourceManager();
+
+    private volatile ResourceType<?>[] resourceTypes = new ResourceType<?>[]{};
+    private final List<Consumer<HasMetadata>> createCallbacks = new CopyOnWriteArrayList<>();
+    private final List<Consumer<HasMetadata>> deleteCallbacks = new CopyOnWriteArrayList<>();
+
+    private static final Map<String, Map<String, Stack<ResourceItem<?>>>> STORED_RESOURCES = new ConcurrentHashMap<>();
+
+    /**
+     * Stores connected kube clients for context
+     *
+     * @param kubeClient kube client
+     * @param cmdClient  cmd client
+     */
+    private record ClusterContext(KubeClient kubeClient, KubeCmdClient<?> cmdClient) {
+    }
 
     private KubeResourceManager() {
-        this.resourceTypes = new ResourceType[]{};
-        client = new KubeClient();
-        if (TestFrameEnv.CLIENT_TYPE.equals(TestFrameConstants.KUBERNETES_CLIENT)) {
-            kubeCmdClient = new Kubectl(kubeClient().getKubeconfigPath());
-        } else {
-            kubeCmdClient = new Oc(kubeClient().getKubeconfigPath());
-        }
+        // Private constructor
     }
 
     /**
-     * Helper class to hold instance
+     * Gets KubeResourceManager instance
+     *
+     * @return singleton instance
      */
-    private static class SingletonHolder {
-        private static final KubeResourceManager INSTANCE = new KubeResourceManager();
+    public static KubeResourceManager get() {
+        return INSTANCE;
     }
 
     /**
-     * Retrieves the singleton instance of KubeResourceManager.
+     * Gets KubeResourceManager instance
      *
-     * @return The singleton instance of KubeResourceManager.
-     * @deprecated Will be removed in future release.
+     * @return singleton instance
      */
-    @Deprecated(since = "release 0.9.0")
+    @Deprecated(since = "0.9.0")
     public static KubeResourceManager getInstance() {
         return get();
     }
 
+    /**
+     * Set the active context for this thread and auto‑restore on close.
+     *
+     * @param id name of cluster context
+     * @return context
+     */
+    public AutoCloseable useContext(String id) {
+        String ctxId = Optional.ofNullable(id).orElse(TestFrameConstants.DEFAULT_CONTEXT_NAME).toLowerCase();
+        if (!CLUSTER_CONFIGS.containsKey(ctxId)) {
+            throw new IllegalArgumentException("Unknown context '" + ctxId +
+                "'. Define env vars with _" + ctxId.toUpperCase());
```

Comment:
```suggestion
                "'. Define env vars with <PREFIX>_" + ctxId.toUpperCase());
```
shouldn't be there something like this?